File: app/src/main/java/com/originb/inkwisenote2/modules/ocr/data/NoteOcrTextDao.java
package com.originb.inkwisenote2.modules.ocr.data;

import androidx.room.*;

import java.util.List;

@Dao
public interface NoteOcrTextDao {
    @Query("SELECT * FROM note_text WHERE note_id = :noteId")
    NoteOcrText readTextFromDb(long noteId);

    @Query("SELECT * FROM note_text")
    List<NoteOcrText> getAllNoteText();

    @Query("SELECT * FROM note_text WHERE extracted_text LIKE '%' || :searchTerm || '%'")
    List<NoteOcrText> searchTextFromDb(String searchTerm);

    @Update
    int updateTextToDb(NoteOcrText noteOcrText);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertTextToDb(NoteOcrText noteRelation);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertTextToDb(List<NoteOcrText> noteRelation);

    @Query("DELETE FROM note_text WHERE note_id = :noteId")
    void deleteNoteText(Long noteId);
}

File: app/src/main/java/com/originb/inkwisenote2/modules/ocr/data/NoteOcrTextDao.kt
package com.originb.inkwisenote2.modules.ocr.data

import androidx.room.*

@Dao
interface NoteOcrTextDao {
    @Query("SELECT * FROM note_text WHERE note_id = :noteId")
    fun readTextFromDb(noteId: Long?): NoteOcrText?

    @get:Query("SELECT * FROM note_text")
    val allNoteText: List<NoteOcrText?>?

    @Query("SELECT * FROM note_text WHERE extracted_text LIKE '%' || :searchTerm || '%'")
    fun searchTextFromDb(searchTerm: String?): List<NoteOcrText?>?

    @Update
    fun updateTextToDb(noteOcrText: NoteOcrText?): Int

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertTextToDb(noteRelation: NoteOcrText?)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertTextToDb(noteRelation: List<NoteOcrText?>?)

    @Query("DELETE FROM note_text WHERE note_id = :noteId")
    fun deleteNoteText(noteId: Long?)
}
